app/db-ls/main.cpp

```diff
@@ -0,0 +1,208 @@
+        break;
+#if QT_VERSION >= 0x050500
+    case QtInfoMsg:
+        if (g_cmdLineOptions.syslogEnabled) {
+            syslog(LOG_INFO, "[INFO] %s\n", qPrintable(msg));
+        } else {
+            fprintf(stderr, "[INFO] %s\n", qPrintable(msg));
+        }
+        break;
+#endif
+    case QtWarningMsg:
+        if (g_cmdLineOptions.syslogEnabled) {
+            syslog(LOG_WARNING, "[WARNING] %s\n", qPrintable(msg));
+        } else {
+            fprintf(stderr, "[WARNING] %s\n", qPrintable(msg));
+        }
+        break;
+    case QtCriticalMsg:
+        if (g_cmdLineOptions.syslogEnabled) {
+            syslog(LOG_CRIT, "[CRITICAL] %s\n", qPrintable(msg));
+        } else {
+            fprintf(stderr, "[CRITICAL] %s\n", qPrintable(msg));
+        }
+        break;
+    case QtFatalMsg:
+        if (g_cmdLineOptions.syslogEnabled) {
+            syslog(LOG_ALERT, "[FATAL] %s\n", qPrintable(msg));
+        } else {
+            fprintf(stderr, "[FATAL] %s\n", qPrintable(msg));
+        }
+        abort();
+    }
+}
+
+void parseCommandLine(QCommandLineParser &parser, QCoreApplication &app, CmdLineOptions *opts)
+{
+    // set defaults
+    opts->debuggingEnabled = false;
+    opts->syslogEnabled = false;
+    opts->sessionBusEnabled = false;
+    opts->key = QString("/");
+
+    parser.setApplicationDescription("openxt simple db storage daemon");
+    parser.addHelpOption();
+    parser.addVersionOption();
+
+    parser.addPositionalArgument("key", QCoreApplication::translate("main", "key"));
+
+    QCommandLineOption debugOption(QStringList() << "d" << "debug",
+                                   QCoreApplication::translate("main", "enable debug/verbose logging"));
+    parser.addOption(debugOption);
+
+    QCommandLineOption syslogOption(QStringList() << "s" << "syslog",
+                                    QCoreApplication::translate("main", "enable logging via syslog"));
+    parser.addOption(syslogOption);
+
+    QCommandLineOption sessionBusOption(QStringList() << "x" << "use-session-bus",
+                                        QCoreApplication::translate("main", "use session bus instead of system bus (useful for testing)"));
+    parser.addOption(sessionBusOption);
+
+    parser.process(app);
+
+    opts->debuggingEnabled = parser.isSet(debugOption);
+    opts->syslogEnabled = parser.isSet(syslogOption);
+    opts->sessionBusEnabled = parser.isSet(sessionBusOption);
+
+    const QStringList posArgs = parser.positionalArguments();
+    if (posArgs.size() >= 1) {
+        opts->key = posArgs.at(0);
+    }
+
+    qDebug() << "debugging enabled:" << opts->debuggingEnabled;
+    qDebug() << "syslog enabled:" << opts->syslogEnabled;
+    qDebug() << "session bus enabled:" << opts->sessionBusEnabled;
+    qDebug() << "key:" << opts->key;
+}
+
+
+// Insane stringify (aka db-ls)
+void lsObject(QMPointer<QMJsonValue> value, QStringList &outStringList, QString key, int level)
+{
+    QString out;
+
+    if (value->isNull()) {
```

Comment:
should always check for null before doing ->. In other words, if(value && value->isNull())